### PR TITLE
docs: update k8s upgrade instructions

### PR DESCRIPTION
**Pull request by lkysow**
With https://github.com/hashicorp/consul-k8s/pull/3000 merged, users can
upgrade their k8s installs using a regular helm upgrade since the
upgrade is now stable.

### CODE REVIEW

File: website/content/docs/k8s/upgrade/index.mdx [hunk 1]
```diff
@@ -115,112 +120,37 @@ to update to the new version. Before you upgrade to a new version:
...
-   and ensuring that all servers are listed and are `alive`.
+## Consul server restarts/upgrades
 
-1. Decrease `updatePartition` by one and upgrade again. Continue until
-   `updatePartition` is `0`. At this point, you may remove the
-   `updatePartition` configuration. Your server upgrade is complete.
+-> *Note for older versions*: For versions of Consul on Kubernetes prior to `1.4.0`, we recommended using the `server.updatePartition` setting to slowly upgrade
+Consul servers. This is no longer necessary due to some changes we made to increase the stability of upgrades, but if you
+are upgrading a version of the chart older than `1.4.0`, use the version drop-down at the top of the page to select a version
+<= `1.17.0` (corresponding to the Consul version in your chart, not the chart version) which contains those instructions.
```
Comment (trujillo-adam) [inline website/content/docs/k8s/upgrade/index.mdx:153]:
```suggestion
Consul servers. Refer to an older version of the documentation for instructions on upgrading to a version of the chart older than `v1.4.0`. Use the version drop-down at the top of this page to select a version older than or equal to `v1.17.0`. Consul documentation versions correspond to the Consul version in your chart, not the chart version, that contains the instructions.
```

Per the style guide, avoid using parentheses and mathematical expressions in prose.

**Review by trujillo-adam — APPROVED** [review]
Thanks for updating. I made a few style guide related suggestions and had one question. Approving so that you can move forward after you've addressed them.

File: website/content/docs/k8s/upgrade/index.mdx [hunk 2]
```diff
@@ -115,112 +120,37 @@ to update to the new version. Before you upgrade to a new version:
...
    ```yaml
    global:
-     image: consul:1.11.2
+     image: "hashicorp/consul:1.11.1"
    ```
 
    </CodeBlockConfig>
 
-2. Determine the version of your existing Helm installation. In this example, version `0.39.0` (from `consul-k8s:0.39.0`) is being used.
+1. Determine the version of your existing Helm installation. In this example, version `0.39.0` (from `consul-k8s:0.39.0`) is being used.
```
Comment (lkysow) [inline website/content/docs/k8s/upgrade/index.mdx:128]:
```suggestion
1. Determine the version of your existing Helm installation. The following example shows that version `0.39.0` is installed. The version is derived from the `CHART` column.
```

**Comment by lkysow** [comment]
Thanks for the quick review @trujillo-adam !

**Pull request merged**